### PR TITLE
[feature](function) support ip function named is_ip_address_in_range(addr, cidr)

### LINKED ISSUE #20584: [Enhancement] Add IP address process function for SQL like ClickHouse
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/doris/issues?q=is%3Aissue) and found no similar issues.

### Description

ClickHouse support some [IP address process function](https://clickhouse.com/docs/zh/sql-reference/functions/ip-address-functions) for SQL. These are very useful for some network service cases.
Hope to add these function ASAP.Thanks.

### Solution

Add IP address process function for SQL like ClickHouse

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by sjyango**
## Proposed changes

Issue Number: close #20584

doris output:
image

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by sjyango** [comment]
run buildall

**Review by zclllyybb — CHANGES_REQUESTED** [review]

File: be/src/vec/functions/function_ip.h [hunk 1]
```diff
@@ -745,4 +740,96 @@ class FunctionIsIPv6String : public IFunction {
...
+                throw Exception(ErrorCode::INVALID_ARGUMENT,
+                                "The arguments of function {} must be String, not NULL",
+                                get_name());
+            }
+            if (nullmap_cidr && (*nullmap_cidr)[i]) {
+                throw Exception(ErrorCode::INVALID_ARGUMENT,
+                                "The arguments of function {} must be String, not NULL",
+                                get_name());
+            }
+            const auto addr = IPAddressVariant(str_addr_column->get_data_at(i).to_string_view());
```
[the remark this answers is not in the document]
Comment (sjyango) [reply be/src/vec/functions/function_ip.h:825]: Thank for reviewing, but I had cast column to concrete column,(see line 782, 786, 792, 796) and `str_addr_column ` `str_cidr_column` had cast to ColumnString whose datatype is `const ColumnString*`.

**Comment by sjyango** [comment]
run buildall

File: be/src/vec/functions/function_ip.h (hunk 1 above)
[the remark this answers is not in the document]
Comment (zclllyybb) [reply be/src/vec/functions/function_ip.h:825]: oh I see. sorry

**Review by zclllyybb — APPROVED** [review]

**Comment by sjyango** [comment]
run pipelinex_p0

File: gensrc/script/doris_builtins_functions.py [hunk 2]
```diff
@@ -2012,11 +2012,12 @@
         [['ipv6_string_to_num_or_default'], 'STRING', ['STRING'], 'ALWAYS_NOT_NULLABLE'],
         [['ipv6_string_to_num_or_null'], 'VARCHAR', ['VARCHAR'], 'ALWAYS_NULLABLE'],
         [['ipv6_string_to_num_or_null'], 'STRING', ['STRING'], 'ALWAYS_NULLABLE'],  
-        [['ipv6numtostring','inet6_ntoa'], 'STRING', ['STRING'], 'ALWAYS_NULLABLE'],
```
Comment (amorynan) [inline gensrc/script/doris_builtins_functions.py:2015]: why delete this ?

**Review by amorynan — APPROVED** [review]
LGTM

**Review by yiguolei — APPROVED** [review]

**Pull request merged**